src/launch/mod.rs

```diff
@@ -261,3 +263,50 @@ pub struct TdxCapabilities {
 
     pub cpuid_configs: Vec<CpuidConfig>,
 }
+
+/// Manually create the wrapper for KVM_MEMORY_ENCRYPT_OP since `kvm_ioctls` doesn't
+/// support `.encrypt_op` for vcpu fds
+use vmm_sys_util::*;
+ioctl_iowr_nr!(
+    KVM_MEMORY_ENCRYPT_OP,
+    kvm_bindings::KVMIO,
+    0xba,
+    std::os::raw::c_ulong
+);
+
+pub struct TdxVcpu {
+    pub fd: kvm_ioctls::VcpuFd,
```

Comment:
```suggestion
    pub fd: &mut kvm_ioctls::VcpuFd,
```

I don't think this struct should consume the `VcpuFd`, as hypervisors will likely have to then clone it once they use the API. Lets keep a reference rather than the `VcpuFd` itself.

This will introduce a lifetime requirement for `TdxVcpu`.